### PR TITLE
Create target

### PR DESCRIPTION
**Pull request by aguskoll**
### What's in

View for crating the target
input validation
Post to the backend to create the target

### Todo

Show the topics list

### How it looks:

https://drive.google.com/file/d/1plJ74jqwIy7dNGMpkXzDguMcdohMjZRO/view?usp=sharing

Screenshot_1611580786

Screenshot_1611581048

### How to test

Run this branch and log in.
Touch the create a target btn and fill data with topic 2
verify that you see a toast with "success creating target".

### CODE REVIEW

File: app/src/main/java/com/rootstrap/android/network/managers/ITargetPointManager.kt [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+import com.rootstrap.android.network.models.TargetPointSerializer
+import com.rootstrap.android.network.models.TopicsSerializer
+import com.rootstrap.android.util.extensions.Data
+
+interface ITargetPointManager {
+
+    suspend fun createTarget(target: Target): Result<Data<TargetPointSerializer>>
+
+    suspend fun getTopics(): Result<Data<TopicsSerializer>>
+}
```
Comment (amaury901130) [inline app/src/main/java/com/rootstrap/android/network/managers/ITargetPointManager.kt]: Totally agree with you, you do not need this, 
managers are more to handle characteristics such as: 
Location, Storage, Database, Preferences, Notifications, Sercvices etc ...
so this should be included in a ViewModel(TargetPointsViewModel) as you suggested.

File: app/src/main/java/com/rootstrap/android/ui/activity/main/targetpoint/CreateTargetView.kt [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+            targetPointsViewModel.createTarget(target)
+        }
+    }
+
+    private fun validateUserInputs(area: Double, title: String?, topic: Int): Boolean =
+        validateArea(area) && validateTargetTitle(title) && validateTopic(topic)
+
+    private fun validateArea(area: Double): Boolean {
+        val isAreaValid = targetPointsViewModel.isAreaValid(area)
+        if (isAreaValid.not())
```
Comment (amaury901130) [inline app/src/main/java/com/rootstrap/android/ui/activity/main/targetpoint/CreateTargetView.kt:48]: 👌

File: app/src/main/res/layout/activity_target_points.xml [hunk 3]
```diff
@@ -1,54 +1,61 @@
...
+                android:textAllCaps="true"
+                android:textColor="@android:color/black"
+                android:textSize="@dimen/text_size_12"
+                android:textStyle="bold" />
+        </LinearLayout>
+    </androidx.constraintlayout.widget.ConstraintLayout>
+
+    <include layout="@layout/fragment_create_target"/>
+
+</androidx.coordinatorlayout.widget.CoordinatorLayout>
```
Comment (amaury901130) [inline app/src/main/res/layout/activity_target_points.xml]: add extra line.

File: app/src/main/res/layout/fragment_create_target.xml [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+        android:layout_marginTop="@dimen/text_input_margin_top"
+        android:text="@string/save_target_btn"
+        android:textAllCaps="true"
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toBottomOf="@id/topic_text_input_layout"
+        app:layout_constraintBottom_toBottomOf="parent"
+        android:layout_marginBottom="@dimen/margin_normal_16"/>
+
+</androidx.constraintlayout.widget.ConstraintLayout>
```
Comment (amaury901130) [inline app/src/main/res/layout/fragment_create_target.xml]: add an extra line

File: app/src/main/java/com/rootstrap/android/network/managers/ITargetPointManager.kt (hunk 1 above)
Comment (aguskoll) [reply app/src/main/java/com/rootstrap/android/network/managers/ITargetPointManager.kt]: I renamed it to Target point service

File: app/src/main/res/layout/fragment_create_target.xml (hunk 4 above)
Comment (aguskoll) [reply app/src/main/res/layout/fragment_create_target.xml]: done

File: app/src/main/res/layout/activity_target_points.xml (hunk 3 above)
Comment (aguskoll) [reply app/src/main/res/layout/activity_target_points.xml]: done

**Review by ximenaperez — APPROVED** [review]

**Pull request merged**